### ISSUE TITLE
[no-unused-modules] Error with ESLint configuration non-standard name

### ISSUE THREAD

**Issue by regseb** [import-js/eslint-plugin-import#2678]
The [`import/no-unused-modules`](https://github.com/import-js/eslint-plugin-import/blob/v2.27.4/docs/rules/no-unused-modules.md) rule doesn't work when the ESLint configuration file doesn't have a standard name. In my projects, I group the configuration files in a directory to avoid having too many files in the root (example with [Cast Kodi](https://github.com/regseb/castkodi/tree/v6.10.0/.metalint)).

- index.js:

  ```JavaScript
  console.log("foo");
  ```

- my_eslint.json:

  ```JSON
  {
      "plugins": ["import"],
      "parserOptions": {
          "sourceType": "module",
          "ecmaVersion": 2022
      },

      "rules": {
          "import/no-unused-modules": [2, { "unusedExports": true }]
      }
  }
  ```

- package.json:

  ```JSON
  {
      "name": "testcase",
      "version": "1.0.0",
      "type": "module",
      "private": true,
      "dependencies": {
          "eslint-plugin-import": "2.27.4",
          "eslint": "8.31.0"
      }
  }
  ```

Run this command: `npx eslint -c my_eslint.json index.js`

```
Oops! Something went wrong! :(

ESLint: 8.31.0

ESLint couldn't find a configuration file. To set up a configuration file for this project, please run:

    npm init @eslint/config

ESLint looked for configuration files in /home/regseb/testcase and its ancestors. If it found none, it then looked in your home directory.

If you think you already have a configuration file or if you need more help, please stop by the ESLint chat room: https://eslint.org/chat/help
```

**Comment by ljharb** [comment]
This seems like an issue with eslint itself, and not with eslint-plugin-import - this plugin doesn't do anything with eslint config.

(typically you'd only ever have one eslint config file in any repo, at the root, and you'd use `overrides` to configure subdirectories, but either way it's specific to eslint)

**Comment by regseb** [comment]
@ljharb I created an issue at ESLint and eslint-plugin-import use indirectly ESLint config:

- https://github.com/eslint/eslint/issues/16787

  > The `import/no-unused-modules` rule indirectly loads eslint configs by [using](https://github.com/import-js/eslint-plugin-import/blob/0778b0390ec18b9fdc1acfd45d10bf1b614d82ff/src/rules/no-unused-modules.js#L21) eslint's internal module `FileEnumerator`. This module isn't intended to be used outside eslint package. When used by eslint, `FileEnumerator` receives full execution context that includes processed CLI options. The error you're getting is caused by the rule not providing the context to the `FileEnumerator` [instance](https://github.com/import-js/eslint-plugin-import/blob/0778b0390ec18b9fdc1acfd45d10bf1b614d82ff/src/rules/no-unused-modules.js#L57-L59) it's using, so it works as if `-c my_eslint.json` is not specified and therefore looks for `.eslintrc.*` config files. We can discuss in https://github.com/eslint/eslint/issues/13481 how this rule could work with the new config system, but until then it seems incompatible with the `-c` usage.

**Comment by ljharb** [comment]
Yes, everything uses it indirectly - the bug is only with something that uses it directly.

**Comment by ljharb** [comment]
Reopening per the comment on the linked issue.

**Comment by tetarchus** [comment]
This also seems to be an issue with the new eslint flat config file. When I enable this rule, I get the message:

```
Oops! Something went wrong! :(

ESLint: 8.38.0

ESLint couldn't find a configuration file. To set up a configuration file for this project, please run:

    npm init @eslint/config

ESLint looked for configuration files in [<rootDir>]/[<childDir>] and its ancestors. If it found none, it then looked in your home directory.

If you think you already have a configuration file or if you need more help, please stop by the ESLint Discord server: https://eslint.org/chat
```

Flat config is in the root directory of the project, with no `-c` flag provided in the eslint command. Without this rule, everything runs fine, but as soon as I enable it, the above error flags up.

**Comment by thenick775** [comment]
+1, still seeing the same on `9.18.0`
